packages/cli/src/main/kotlin/elide/tool/cli/cmd/repl/ToolShellCommand.kt

```diff
@@ -347,26 +350,10 @@ import org.graalvm.polyglot.Engine as VMEngine
     )
     internal var wasm: Boolean = false
 
-    /** Apply configuration to the JS VM based on the provided arguments. */
-    override fun apply(): Stream<out VMProperty> {
-      return listOfNotNull(
-        // set strict mode
-        VMStaticProperty.active("js.strict"),
-
-        // set ECMA version
-        VMStaticProperty.of("js.ecmascript-version", if (ecma.name.startsWith("ES")) {
-          ecma.name.substring(2)
-        } else {
-          ecma.name
-        }),
-
-        // if WASM is enabled, pass a prop for it
-        if (wasm) {
-          VMStaticProperty.active("js.webassembly")
-        } else {
-          null
-        }
-      ).stream()
+    /** Apply these settings to the configuration for the JavaScript runtime plugin. */
+    internal fun apply(config: JavaScriptConfig) {
```

Comment:
Instead of returning the property stream, the options are applied to the `JavaScript` engine plugin's configuration. This is used in the `configureEngine` event, when the JS plugin is installed.